src/main/java/com/github/zly2006/reden/rvc/tracking/client/ClientTracking.kt

```diff
@@ -0,0 +1,50 @@
+package com.github.zly2006.reden.rvc.tracking.client
+
+import com.github.zly2006.reden.render.BlockBorder
+import com.github.zly2006.reden.rvc.gui.selectedStructure
+import com.github.zly2006.reden.rvc.tracking.TrackedStructure
+import com.github.zly2006.reden.utils.handToolItem
+import fi.dy.masa.malilib.event.InputEventHandler
+import fi.dy.masa.malilib.hotkeys.IMouseInputHandler
+import net.minecraft.client.MinecraftClient
+import net.minecraft.util.hit.BlockHitResult
+import net.minecraft.util.hit.HitResult
+import org.lwjgl.glfw.GLFW
+
+fun registerSelectionTool() {
+    InputEventHandler.getInputManager().registerMouseInputHandler(object : IMouseInputHandler {
+        override fun onMouseClick(mouseX: Int, mouseY: Int, eventButton: Int, eventButtonState: Boolean): Boolean {
+            if (!eventButtonState) return false // ensure mouse down
+            val mc = MinecraftClient.getInstance()
+            if (mc.currentScreen != null) return false // ensure no gui
+            mc.player?.handToolItem?.takeIf { it } ?: return false // ensure hand tool item
+
```

Comment:
Suggest adding gamemode checks. You won't want this to trigger in survival.

Also this line can be simply written as an if statement.

Reply:
i think this should can be triggered in survival